ui/packages/catalog-ui-search/src/main/webapp/react-component/utils/validation/validation.tsx

```diff
@@ -38,31 +52,339 @@ export function showErrorMessages(errors: any) {
+    </Invalid>
+  ) : null
+}
+
+export function validateListOfPoints(coordinates: any[], mode: string) {
+  let message = ''
+  const isLine = mode.includes('line')
+  const numPoints = isLine ? 2 : 4
+  if (
+    !mode.includes('multi') &&
+    !coordinates.some(coords => coords.length > 2) &&
+    coordinates.length < numPoints
+  ) {
+    message = `Minimum of ${numPoints} points needed for ${
+      isLine ? 'Line' : 'Polygon'
+    }`
+  }
+  coordinates.forEach(coordinate => {
+    if (coordinate.length > 2) {
+      coordinate.forEach((coord: any) => {
+        if (hasPointError(coord))
+          message = JSON.stringify(coord) + ' is not a valid point.'
       })
+    } else {
+      if (mode.includes('multi')) {
+        message = `Switch to ${isLine ? 'Line' : 'Polygon'}`
+      } else if (hasPointError(coordinate)) {
+        message = JSON.stringify(coordinate) + ' is not a valid point.'
+      }
     }
+  })
+  return { error: message.length > 0, message }
+}
+
+export function validateLinePolygon(currentValue: string, mode: string) {
+  if (!is2DArray(currentValue)) {
+    return { error: true, message: 'Not an acceptable value' }
+  }
+  try {
+    const pointsValid = validateListOfPoints(JSON.parse(currentValue), mode) 
+    return pointsValid === undefined ? initialErrorState : pointsValid
+  } catch (e) {
+    return initialErrorState
+  }
+}
+
+export const initialErrorState = {
+  error: false,
+  message: '',
+}
+
+export const initialErrorStateWithDefault = {
+  error: false,
+  message: '',
+  defaultValue: '',
+}
+
+function is2DArray(coordinates: string) {
+  try {
+    const parsedCoords = JSON.parse(coordinates)
+    return Array.isArray(parsedCoords) && Array.isArray(parsedCoords[0])
+  } catch (e) {
+    return false
+  }
+}
+
+function hasPointError(point: any[]) {
+  if (
+    point.length !== 2 ||
+    (Number.isNaN(Number.parseFloat(point[0])) &&
+      Number.isNaN(Number.parseFloat(point[1])))
+  ) {
+    return true
+  } else if (
+    point[0] > 180 ||
+    point[0] < -180 ||
+    point[1] > 90 ||
+    point[1] < -90
+  ) {
+    return true
+  }
+  return false
+}
+
+function getGeometryErrors(filter: any): Set<string> {
+  const geometry = filter.geojson && filter.geojson.geometry
+  const bufferWidth =
+    filter.geojson.properties.buffer && filter.geojson.properties.buffer.width
+  const errors = new Set<string>()
+  if (!geometry) {
+    return errors
+  }
+  switch (filter.geojson.properties.type) {
+    case 'Polygon':
+      if (!geometry.coordinates[0].length || geometry.coordinates[0].length < 4) {
+        errors.add(
+          'Polygon coordinates must be in the form [[x,y],[x,y],[x,y],[x,y], ... ]'
+        )
+      }
+      break
+    case 'LineString':
+      if (!geometry.coordinates.length || geometry.coordinates.length < 2) {
+        errors.add('Line coordinates must be in the form [[x,y],[x,y], ... ]')
+      }
+      if (!bufferWidth || bufferWidth == 0) {
```

Comment:
```suggestion
      if (!bufferWidth || bufferWidth === 0) {
```
also, is this second check necessary?

Reply:
also, didn't we talk about checking for negative values here as well @andrewzimmer @willwill96 ?